R/mrgindata.R

```diff
@@ -81,45 +81,62 @@ convert_character_cmt <- function(data, mod) {
   return(data)
 }
 
-signal_drop <- function(dm, x, to_signal, context) {
-  drop <- setdiff(names(x), dimnames(dm)[[2]])
-  drop <- intersect(drop, to_signal)
-  for(d in drop) {
+check_dropped_cols <- function(dm, x, check, context) {
+  new_col_set <- dimnames(dm)[[2]]
+  old_col_set <- names(x)
+  drop <- old_col_set[!old_col_set %in% new_col_set]
+  drop <- drop[drop %in% check]
+  if(!length(drop)) return(invisible(NULL))
+  body <- vector(mode = "character", length = length(drop))
+  names(body) <- rep("x", length(body))
+  for(i in seq_along(drop)) {
+    d <- drop[i]
     type <- paste0(class(x[[d]]), collapse = ",")
-    msg <- c(context, " dropped column: ", d, " (", type, ")")
-    message(msg)  
+    body[i] <- paste0(context, " column: ", d, " (", type, ")")
   }
-  invisible(NULL)
+  abort(
+    message = "Found input data that cannot be used for simulation", 
+    body = body, 
+    call = caller_env()
+  )
 }
 
 ##' Validate and prepare a data sets for simulation
 ##'
-##' This function is called by mrgsim.  Users may also call this function
-##' to pre-validate data when the same data set is used for repeated 
-##' simulation.
+##' This function is called by [mrgsim()] and friends.  Users may also call 
+##' this function to pre-validate data when the same data set is used for 
+##' repeated simulation.
 ##'
-##' @param x data.frame or matrix
-##' @param m a model object
-##' @param verbose logical
-##' @param quiet if \code{TRUE}, messages will be suppressed
+##' @param x data.frame or matrix.
+##' @param m a model object.
+##' @param verbose logical.
+##' @param quiet if `TRUE`, messages will be suppressed.
+##' 
+##' @details
+##' An error will be issued when
+##' - non-numeric data is found in columns sharing names with model parameters
+##' - non-numeric data is found in reserved data items related to dosing 
+##'   (see `mrgsolve:::GLOBALS$CARRY_TRAN`)
```

Comment:
Is referencing this internal `GLOBALS` data structure a bad idea? I don't want to have to maintain this list off the contents of what is in there but also feels odd to expose this internal data.

Reply:
In my opinion it's okay to reference.  It's useful to those that want to see the list, but it being internal should make it clear that it shouldn't be relied on in scripts, etc.